src/handler.js

```diff
@@ -47,6 +46,59 @@ const createProxy = apiKey => {
   return proxy;
 };
 
+const match = route('/:version/:tableName/:recordId?');
+const parseUrl = (originalUrl, airtableBaseId) => {
+  const components = parse(originalUrl);
+  const params = match(components.pathname);
+
+  if (params === false) {
+    const originalPath = components.path;
+    return { url: originalPath, tableName: false };
+  }
+
+  const proxyUrl =
+    '/' +
+    compact([
+      params.version,
+      airtableBaseId,
+      params.tableName,
+      params.recordId,
+      components.search,
+    ]).join('/');
+
+  return {
+    proxyUrl,
+    tableName: params.tableName,
+  };
+};
+
+const allMethods = ['GET', 'PUT', 'POST', 'PATCH', 'DELETE'];
+const getAllowedMethods = (config, tableName) => {
+  let allowedMethods = [];
+
+  if (!tableName) {
+    allowedMethods = allMethods;
+  } else {
+    const hasRouteSpecificConfig = isObject(config.allowedMethods);
+    const configAllowedMethods = hasRouteSpecificConfig
+      ? config.allowedMethods[tableName] || []
+      : config.allowedMethods;
+
+    allowedMethods =
+      configAllowedMethods === '*' ? allMethods : configAllowedMethods;
+  }
+
+  return ['OPTIONS', ...allowedMethods];
+};
+
+const isAllowed = (allowedMethods, method) => {
+  if (Array.isArray(permissions) && permissions.includes(method)) {
```

Comment:
I'm getting an undefined error here. I think you meant:

```
  if (Array.isArray(allowedMethods) && allowedMethods.includes(method)) {
```

Reply:
Sorry 'bout that. I made some tweaks after doing a pass of testing and forgot to update this line. Just pushed a fix.